frontend/public/components/utils/selector-input.jsx

```diff
@@ -23,7 +23,16 @@ export class SelectorInput extends React.Component {
     };
   }
 
+  componentDidUpdate(prevProps) {
+    if (!_.isEqual(prevProps.tags, this.props.tags)) {
+      this.setState({ tags: this.props.tags });
+    }
+  }
   static arrayify(obj) {
+    if (Array.isArray(obj)) {
```

Comment:
It would be better to pass in the values in the format the component expects rather than trying to handle different formats here.